wrangler-core/src/main/java/io/cdap/directives/column/Keep.java

```diff
@@ -93,4 +103,18 @@ public Mutation lineage() {
     keep.forEach(column -> builder.relation(column, column));
     return builder.build();
   }
+  @Override
+  public Relation transform(RelationalTranformContext relationalTranformContext,
+                            Relation relation) {
+    Optional<ExpressionFactory<String>> expressionFactory = getExpressionFactory(relationalTranformContext);
+    if (!expressionFactory.isPresent()) {
+      return new InvalidRelation("Cannot find an Expression Factory");
+    }
+    Map<String, Expression> keepCol = generateColumnExpMap(keep.stream().collect(Collectors.toList()),
```

Comment:
To avoid having to stream and collect a set to a list, you can modify the function to take a collection as an argument. There are common methods to iterate over a collection